src/Pos/Block/Logic.hs

```diff
@@ -458,7 +458,7 @@ toTxpBlock = bimap convertGenesis convertMain
 -- header hash of new tip. It's up to caller to log warning that
 -- partial application happened.
 verifyAndApplyBlocks
-    :: (WorkMode ssc m, SscWorkersClass ssc)
+    :: (MonadDBCore m, WorkMode ssc m, SscWorkersClass ssc)
```

Comment:
Yes, actually I don't like `WorkMode` as it is now and I think it should be reworked.